iib/workers/tasks/utils.py

```diff
@@ -24,6 +24,136 @@
+    :param dict binary_image_config: the dict of config required to identify the appropriate
+        ``binary_image`` to use.
+    :return: pull specification of the binary_image to be used for this build.
+    :rtype: str
+    :raises IIBError: when the config value for the ocp_version and distribution_scope is missing.
+    """
+    binary_image = binary_image_config.get(distribution_scope, {}).get(ocp_version, None)
+    if not binary_image:
+        raise IIBError(
+            'IIB does not have a configured binary_image for'
+            f' distribution_scope : {distribution_scope} and ocp_version: {ocp_version}.'
+            ' Please specify a binary_image value in the request.'
+        )
+
+    return binary_image
+
+class RequestConfig:
+    """
+    Request config abstract class.
+    :param binary_image:  the pull specification of the container image
+                          where the opm binary gets copied from.
+    :type binary_image: str
+    :param overwrite_from_index_token: the token used for overwriting the input
+        ``from_index`` image. This is required for
+        non-privileged users to use ``overwrite_from_index``.
+        The format of the token must be
+        in the format "user:password".
+    :type overwrite_from_index_token: str
+    :param distribution_scope: the scope for distribution
+        of the index image, defaults to ``None``.
+    :type distribution_scope: str
+    :param source_from_index: the pull specification of the container image
+        containing the index that will be used
+        as a base of the merged index image.
+    :type source_from_index: str
+    :param target_index: the pull specification of the container image
+        containing the index whose new data will be added
+        to the merged index image.
+    :type target_index: str
+    :param binary_image_config: the dict of config required to
+        identify the appropriate ``binary_image`` to use.
+    :type binary_image_config: dict
+    """
+
+    _attrs = [
+        "binary_image",
+        "overwrite_from_index_token",
+        "add_arches",
+        "distribution_scope",
+        "source_from_index",
+        "target_index",
+        "binary_image_config",
+    ]
+    __slots__ = _attrs
+
+    def __init__(self, **kwargs):
+        """
+        Request config __init__.
+        Do not use this directly, use subclasses instead.
+        :Keyword Arguments:
+            See `_attrs` to check accepted keyword arguments.
+        """
+        for key in self.__slots__:
+            if key in kwargs:
+                setattr(self, key, kwargs[key])
+            else:
+                setattr(self, key, None)
+
+    def __eq__(self, other):
+        if type(self) == type(other) and [getattr(self, x) for x in self.__slots__] == [
+            getattr(self, x) for x in self.__slots__
+        ]:
+            return True
+        return False
+
+    def get_binary_image(self, index_info, distribution_scope):
```

Comment:
well, it cannot be property because it requires some attributes to determine binary image if it's not set. But I can make binary_image -> _binary_image and only available public method get_binary_image() which will return _binary_image or get binary base on attributes